src/t_hash.c

```diff
@@ -2631,10 +2684,32 @@ void hmgetCommand(client *c) {
     if (expired) {
         notifyKeyspaceEvent(NOTIFY_HASH, "hexpired", c->argv[1], c->db->id);
         if (deleted)
-            notifyKeyspaceEvent(NOTIFY_GENERIC, "del", c->argv[1], c->db->id); 
+            notifyKeyspaceEvent(NOTIFY_GENERIC, "del", c->argv[1], c->db->id);
     }
 }
 
+/* Helper function to build canonical HDEL command for HGETDEL propagation */
+static void buildCanonicalHdelCommand(client *c, int key_pos, int first_field_pos, int field_count) {
```

Comment:
it's similar as hgetex, if a method is not reusable, extracting it is not valuable.